python/kfserving/kfserving/storage.py

```diff
@@ -31,10 +32,15 @@ class Storage(object): # pylint: disable=too-few-public-methods
     @staticmethod
     def download(uri: str, out_dir: str = None) -> str:
         logging.info("Copying contents of %s to local", uri)
+        
+        is_local = False
         if uri.startswith(_LOCAL_PREFIX) or os.path.exists(uri):
-            return Storage._download_local(uri)
+            is_local = True            
 
         if out_dir is None:
+            if is_local:
+                # noop if out_dir is not set and the path is local
+                return Storage._download_local(uri)
```

Comment:
we'd better to check if out_dir is a valid directory and throw correct exception if it's invalid.